acestep/core/generation/handler/generate_music_execute.py

```
@@ -41,33 +93,34 @@ def _run_generate_music_service_with_progress(
+            if gen_thread.is_alive():
+                logger.error(
+                    f"[generate_music] service_generate exceeded {_DEFAULT_GENERATION_TIMEOUT}s "
+                    f"timeout (batch={actual_batch_size}, steps={inference_steps}, "
+                    f"duration={audio_duration}s).  The CUDA operation may still be "
+                    f"running in the background."
+                )
+                raise TimeoutError(
+                    f"Music generation timed out after {_DEFAULT_GENERATION_TIMEOUT} seconds.  "
+                    f"This usually means the GPU ran out of VRAM or the diffusion loop "
+                    f"stalled.  Try reducing batch size, duration, or inference steps."
+                )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)